src/server/main_service.cc

```diff
@@ -1057,7 +1057,10 @@ std::optional<ErrorReply> Service::VerifyCommandState(const CommandId* cid, CmdA
       allowed_by_state = false;
       break;
     case GlobalState::TAKEN_OVER:
-      allowed_by_state = !cid->IsWriteOnly();
+      // Only Admin commands are allowed
+      // we prohibit even read commands, because read commands running in pipeline can take a while
+      // to send all data to a client which leads to fail in takeover
+      allowed_by_state = dfly_cntx.conn()->IsPrivileged() || cid->name() == "DFLY";
```

Comment:
DFLY is admin command
CO::ADMIN

you write in comment only admin commands are allowed but the check dfly_cntx.conn()->IsPrivileged() is if the command is from admin port

So maybe just check command flag CO::ADMIN

Reply:
Not so simple, but it's a good idea I will use it